### ISSUE TITLE
Subscription with filter is not released when client diconnects

### ISSUE THREAD

**Issue by ivibe** [nestjs/graphql#309]
## I'm submitting a...
<pre><code>[x] Bug report</code></pre>

## Current behavior
When client disconnects from server subscription is not released.

## Expected behavior
Subscription must be released as soon as client disconnects.

## Minimal reproduction of the problem with instructions
Let's take the sample 12-graphql-apollo from nestjs/nest

```typescript
const pubSub = new PubSub();

@Resolver('Cat')
export class CatsResolvers {
  constructor() {
    setTimeout(() => 
      pubSub.publish('catCreated', {
        catCreated: { id: 2, name: 'Cat2', age: 3 }
      })
    , 10000);
  }

  @Subscription('catCreated', {
    filter: () => {
      console.log('Subcription filter');
      return true;
    },
    resolve: ({ catCreated }) => {
      console.log('Subcription resolve');
      return catCreated;
    }
  })
  catCreated() {
    return pubSub.asyncIterator('catCreated');
  }
}
```

If withtin 10 seconds after server started we execute and stop next query n times in the playground 
```graphql
subscription catCreated {
  catCreated {
    name
    age
  }
}
```
in console we'll see n messages 
'Subcription filter' and 'Subcription resolve'

If we comment out the filter part of Subscription decorator, no matter how many times the subcription query will be executed, there will be only one message in console
'Subcription resolve'

## What is the motivation / use case for changing the behavior?
Current behavior leads to memory leak.

## Environment

<pre><code>
Nest version: 6.3.2
Nest GraphQL version: 6.2.4
 
For Tooling issues:
- Node version: 12.4.0
- Platform:  Windows
</code></pre>

**Comment by LukaszSkwara** [comment]
![image](https://user-images.githubusercontent.com/17692845/61049543-0ba96580-a3e5-11e9-88f9-2f3d18972da5.png)
Memory usage. Downgrade at 11:09 to nest 5.7.3.
![image](https://user-images.githubusercontent.com/17692845/61049865-d5b8b100-a3e5-11e9-92df-baa052dd9264.png)

**Comment by kamilmysliwiec** [comment]
Fixed in 6.3.0. Thanks for reporting!
